### PR TITLE
Initial changes for dashboards automation

### PR DESCRIPTION
**Pull request by vishnuchalla**
## Type of change
Added: Support for multiple multiple folders, python client
Modified: Internal logic for creating/updating dashboards
Deleted: Removed old logic as it is not flexible for automation

- [x] Refactor
- [x] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

As an initial step in our plan to automating dashboards. Adding support for multiple folders and integrating python client for ease of maintenance.

## Related Tickets & Documents

- Related Issue # https://github.com/cloud-bulldozer/performance-dashboards/issues/77

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Tested in the local environment, using the same old methodology.
* `make` to generate the `rendered` folder
* Built syncer image following the steps in README.md
* And finally ran `dittybopper/syncer/deploy.sh`.

### CODE REVIEW

File: .github/workflows/grafana.yml [hunk 1]
```diff
@@ -25,13 +21,10 @@ jobs:
...
-        dashboard_list=($(echo $DASHBOARDS));
-        for path in "${dashboard_list[@]}"; do
-        full_path="rendered/${path}";
-        echo "Importing ${full_path}";
-        dashboard=$(cat ${full_path});
+        for t in rendered/**/*.json; do
+        echo "Importing ${t}";
+        dashboard=$(cat ${t});
         echo "{\"dashboard\": ${dashboard}, \"overwrite\": true}" |
         curl -k -Ss -XPOST -H "Content-Type: application/json" -H "Accept: application/json" -d@-
```
Comment (rsevilla87) [inline .github/workflows/grafana.yml:28]: The internal grafana won't be accessible from outside

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu
```
Comment (rsevilla87) [inline Dockerfile:1]: We're migrating from ubi-minimal to ubuntu, I don't see it very optimal....

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+WORKDIR /performance-dashboards
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Install necessary libraries for subsequent commands
+RUN apt-get update && apt-get install -y podman dumb-init python3.6 python3-distutils python3-pip python3-apt
```
Comment (rsevilla87) [inline Dockerfile:7]: Why do we need all of this?, only python3 and python3-requirements should be necessary

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+RUN chmod -R 775 /performance-dashboards
+
+# Install dependencies
+RUN python3 -m pip install --upgrade pip
+RUN pip install -r requirements.txt
+
+# Cleanup the installation remainings
+RUN apt-get clean autoclean && \
+    apt-get autoremove --yes && \
+    rm -rf /var/lib/{apt,dpkg,cache,log}/
```
Comment (rsevilla87) [inline Dockerfile:19]: These lines are not necessary and don't make any effect, as the layer from apt cache is still part of the container image.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -17,6 +17,19 @@ on:
...
+
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: make deps
+
+    - name: Run jsonnetfmt
+      run: for t in templates/**/*.jsonnet; do echo "Testing template ${t}"; ./bin/jsonnetfmt --test $t; echo 'Results:' ${?}; done
```
Comment (rsevilla87) [inline .github/workflows/ci.yml:31]: We can update the `format` target from the makefile to include some comments so we can reduce this expression to:

```suggestion
    - name: Get dependencies
      run: make format
```

File: .github/workflows/grafana.yml (hunk 1 above)
Comment (vishnuchalla) [reply .github/workflows/grafana.yml:28]: Yes, Those actions in workflows are not of any use right now. They just help us to do some sanity checks, if we are able to build jsons and deploy them into grafana. To start with lets keep it that way and our plan is to extend the same actions for our automatic deployments in future.

File: Dockerfile (hunk 2 above)
Comment (vishnuchalla) [reply Dockerfile:1]: using ubi-minimal as suggested.

File: Dockerfile (hunk 3 above)
Comment (vishnuchalla) [reply Dockerfile:7]: Removed unnecessary dependencies and kept only minimal set of packages.

File: Dockerfile (hunk 4 above)
Comment (vishnuchalla) [reply Dockerfile:19]: Removed them as they do not have any significant affect in this case.

File: .github/workflows/grafana.yml (hunk 1 above)
Comment (rsevilla87) [reply .github/workflows/grafana.yml:28]: But this workflow is gonna fail, is that expected?

**Review by rsevilla87 — APPROVED** [review]
lgtm!

**Pull request merged**